init/luks.go

```diff
@@ -370,7 +393,13 @@ func luksOpen(dev string, mapping *luksMapping) error {
 		}
 	}
 	if len(checkSlotsWithPassword) > 0 {
-		go requestKeyboardPassword(volumes, d, checkSlotsWithPassword, mapping.name)
+		// is there a keyfile defined for the password for this volume?
+		if len(mapping.keyfile) > 0 {
```

Comment:
If the keyfile is specified do we expect that at least one LUKS slot could be unlocked with it? In this case, do we still want to request password using keyboard?

Reply:
A keyfile is just a file containing a passphrase. Often these are large (4k of random data), but they are done in addition to an actual password in another slot. The keyfile is for unsealing without manual intervention (avoiding the 2nd round of entering a password for instance), however should that keyfile become lost or damaged the original password is still usable. The keyfile and passwords use the same slot types, so in this case I am giving them the opportunity to manually unseal and finish booting using a password if keyfile fails so they can fix the issue (maybe they forgot to add the file to initramfs, lost the thumbdrive holding it/etc) instead of just locking them completely out of the computer.